libmscore/score.cpp

```diff
@@ -1880,6 +1880,10 @@ void Score::scanElementsInRange(void* data, void (*func)(void*, Element*), bool
                         ss->scanElements(data, func, all);
                         }
                   }
+            else if (e->isSpannerSegment()) {
+                  const auto ss = toSpannerSegment(e);
+                  ss->scanElements(data, func, all);
+                  }
```

Comment:
See corresponding code in master, https://github.com/musescore/MuseScore/blob/e9880db635f7f13a90e33f171f4b21a8a56b395c/src/engraving/dom/score.cpp#L2060-L2071

Reply:
Using a set like that doesn't make much sense to me. Does it to you? If so please explain. Almost seems like an extra unnecessary step.

Reply:
No idea really, but seems needed on order to process spanners in the right order?

Reply:
It's still within the loop though and only acts on the most recently added spanner. Doesn't make any sense!
Plus it almost seems superfluous to go through all the segments. Continued segments already are included via range selection from what I can tell

<img width="1334" height="394" alt="image" src="https://github.com/user-attachments/assets/dd7f2fe0-8be0-4e9f-ab1f-a1c71cb56e84" />


For example that blue shows the range selection, yet the "continued" slur is selected there and so included with the filtering

but again... i suppose sticking with the same code for backporting purposes is the main idea

Reply:
> but again... i suppose sticking with the same code for backporting purposes is the main idea

Yers, it is. Not sure though whether it is feasible here, that `insert()` seems to be C++17, not sure we can rely in this being avialble for all current platforms and builds?

Reply:
I'll try it and do a force push to see if it goes through and see what the conveyor says

Reply:
Would this work?
```cpp
      for (Element* e : _selection.elements()) {
            if (!e->isSpannerSegment())
                  continue;
            Spanner* spanner = toSpannerSegment(e)->spanner();
            for (SpannerSegment* ss : spanner->spannerSegments()) {
                  ss->scanElements(data, func, all);
                  }
            }
```

Reply:
It's not really any different than what I was doing though initially though, save for using a continue instead of an if-clause. Quite literally the same thing.